test/prop_pobox.erl

```diff
@@ -0,0 +1,111 @@
+-module(prop_pobox).
+-include_lib("proper/include/proper.hrl").
+-export([
+    prop_we_always_go_to_passive_mode_after_an_automatic_transfer/0,
+    prop_we_always_go_to_passive_mode_after_a_give_away_transfer/0,
+    prop_will_discard_after_max_size/0
+]).
+
+-type pobox_max_size() :: pos_integer().
+-type pobox_initial_state() :: passive | notify.
+-type crash_vector() :: pos_integer().
+-type give_away_vector() :: pos_integer().
+-type message() :: {msg, binary()} | {resize, pos_integer()} | usage.
+-type messaging_strategy() :: async | sync | info.
+-type messaging_behaviour() :: {non_neg_integer(), messaging_strategy(), message()}.
+-type automatic_behaviour() :: {wait, non_neg_integer()} | notify | {active, non_neg_integer() | all}.
+-type give_away_behaviour() :: {wait, non_neg_integer()} | notify | {active, non_neg_integer() | all}.
+-type buffer_type() :: queue | keep_old | stack | {mod, pobox_queue_buf}.
+-type automatic_scenario() :: {crash_vector(), list(automatic_behaviour()), list(messaging_behaviour()), buffer_type()}.
+-type give_away_scenario() :: {give_away_vector(), list(give_away_behaviour()), list(messaging_behaviour()), buffer_type()}.
+-type max_size_scenario() :: {pobox_max_size(), buffer_type(), pobox_initial_state()}.
+
+prop_we_always_go_to_passive_mode_after_an_automatic_transfer() ->
+    %% Random message sequence
+    %% Random behaviour sequence
+    %% Crash at random time vector
+    %% pobox should be in passive mode
+    erlang:process_flag(trap_exit, true),
+    ?FORALL(_Scenario = {CrashVector, Behaviours, MessagingBehaviours, BufType}, automatic_scenario(), begin
+        {ok, SCPid} = pobox_state_changer:start_link(self(), Behaviours, CrashVector, BufType),
+        Box = pobox_state_changer:get_pobox(SCPid),
+        %io:format("~p~n", [Scenario]),
+        Procs = [erlang:spawn_opt(node(), fun() -> catch do_messaging_behaviour(Box, MessagingBehaviour) end, [monitor])
```

Comment:
you can just use `spawn_monitor(fun() -> ... end)` and generally wouldn't need to `catch` anything since you're looking ofr things going down.

Reply:
Alright, consider it done.